### PR TITLE
[Feature][ClickhouseFile] Support add publicKey to identity

### PR DESCRIPTION
**Pull request by caicancai**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by caicancai** [comment]
cc @Hisoka-X

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/ClickhouseConfig.java [hunk 1]
```diff
@@ -167,6 +167,12 @@ public class ClickhouseConfig {
                     .defaultValue(Collections.emptyMap())
                     .withDescription("Clickhouse custom config");
 
+    public static final Option<String> KEY_PATH =
+            Options.key("rsync_key_path")
+                    .stringType()
+                    .defaultValue(null)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/ClickhouseConfig.java]:
```suggestion
                    .noDefaultValue()
```

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/ClickhouseConfig.java [hunk 2]
```diff
@@ -167,6 +167,12 @@ public class ClickhouseConfig {
...
+                    .withDescription("The path of rsync key file");
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/ClickhouseConfig.java]:
```suggestion
                    .withDescription("The path of rsync/ssh key file");
```

**Review by Hisoka-X — COMMENTED** [review]
Please update https://github.com/apache/seatunnel/blob/eb0bf891ce19ef04ee0fb67878e3ca17cf2ff488/seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/file/ClickhouseFileSinkFactory.java#L51 too.

**Comment by caicancai** [comment]
> Please update
> 
> https://github.com/apache/seatunnel/blob/eb0bf891ce19ef04ee0fb67878e3ca17cf2ff488/seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/file/ClickhouseFileSinkFactory.java#L51
> 
> too.

done

**Review by Hisoka-X — DISMISSED** [review]

**Comment by wuchunfu** [comment]
@caicancai The document seems to have not been updated. Please update the document. Thanks

**Comment by caicancai** [comment]
> @caicancai The document seems to have not been updated. Please update the document. Thanks

done, There are many vague and wrong places in the ClickhouseFile documentation. I guess I will submit a PR to improve it later.

**Comment by Hisoka-X** [comment]
> > @caicancai The document seems to have not been updated. Please update the document. Thanks
> 
> done, There are many vague and wrong places in the ClickhouseFile documentation. I guess I will submit a PR to improve it later.

Thanks @caicancai

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**